cmd/goal/node.go

```diff
@@ -174,18 +191,27 @@ var catchupCmd = &cobra.Command{
 				if err != nil {
 					reportErrorf(errorCatchpointLabelMissing, errorUnableToLookupCatchpointLabel, err.Error())
 				}
-				args = append(args, label)
-				if !fastCatchupForce {
-					fmt.Printf(nodeConfirmImplicitCatchpoint, label)
-					reader := bufio.NewReader(os.Stdin)
-					text, _ := reader.ReadString('\n')
-					text = strings.Replace(text, "\n", "", -1)
-					if text != "yes" {
-						reportErrorf(errorAbortedPerUserRequest)
-					}
+				catchpoint = label
+			}
+
+			// Prompt user to confirm implicit catchpoint.
+			if !fastCatchupForce && len(args) == 0 {
+				fmt.Printf(nodeConfirmImplicitCatchpoint, catchpoint)
+				reader := bufio.NewReader(os.Stdin)
+				text, _ := reader.ReadString('\n')
+				text = strings.Replace(text, "\n", "", -1)
+				if text != "yes" {
+					reportErrorf(errorAbortedPerUserRequest)
```

Comment:
If you move the "implicit check" up to line 194, where you already know the label has been looked up implicitly, you won't need the `len(args) == 0` predicate.  And it seems reasonable to do the implicit check in the place where you decided to do the implicit lookup.

Reply:
Nice. I wanted to put it outside the `OnDataDirs` iterator with the other arg length check to avoid the second length check. Missed this logic.

Reply:
This suggestion has been implemented.